tests/ut/py/test_worker/test_orphan_child_reaping.py

```
@@ -0,0 +1,97 @@
+    except PermissionError:
+        return True
+    return True
+
+
+@pytest.mark.skipif(not sys.platform.startswith("linux"), reason="uses pgrep -P to enumerate forked children")
+def test_children_exit_when_parent_is_killed(tmp_path):
+    out_path = tmp_path / "children.txt"
+    reported: list[int] = []
+    try:
+        with out_path.open("w") as out:
+            parent = subprocess.Popen([sys.executable, "-c", _PARENT_SRC % NUM_SUB_WORKERS], stdout=out, stderr=out)
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)